server/src/app.js

```diff
@@ -33,29 +34,30 @@ app.use(
       "Authorization",
       "Content-Length",
       "X-Requested-With",
-      "encoded-compressed-query",
+      "gql-query",
     ],
   })
 );
 
 app.use(function (req, res, next) {
   res.header("cache-control", "public, max-age=31536000");
 
-  // Often want to use GET to leverage http caching, but some of out longer queries are too long for a query parameter
-  // Instead, the client makes a GET with a header containing a compressed and base 64 encoded copy of the query (the query parameter becomes a hash, for cache busting)
-  // In that case, we mutate the request here to recover the query and let the server pretend it received a normal POST
-  if (
-    req.method === "GET" &&
-    !_.isEmpty(req.headers["encoded-compressed-query"])
-  ) {
-    console.log(`Request type: ${req.originalUrl}, GET with compressed query`);
-    convert_GET_with_compressed_query_to_POST(req); // mutates req, changes made persist to subsequent middleware
+  // Often want to use GET to leverage http caching, but query batching is not supported by GET parameters
+  // Instead, the client makes a GET with a header containing the query
+  // For caching, we also add a hash of the query as a GET parameter
+  // Since apollo isn't expecting this type of request, we mutate it to make it look like a normal POST request
+  if (req.method === "GET" && !_.isEmpty(req.headers["gql-query"])) {
+    console.log(`Request type: ${req.originalUrl}, GET with query header`);
+    convert_GET_with_query_to_POST(req); // mutates req, changes made persist to subsequent middleware
   } else {
     console.log(`Request type: ${req.originalUrl}, ${req.method}`);
   }
 
-  process.env.USE_REMOTE_DB &&
-    console.log(JSON.stringify(get_log_object_for_request(req)));
+  if (process.env.USE_REMOTE_DB) {
+    get_log_objects_for_request(req).forEach((log_obj) => {
+      console.log(JSON.stringify(log_obj));
+    });
```

Comment:
Since a request can now contain multiple queries, we log each query. Part of the log objects is a batch_hash, which can be used to identify which query logs were part of the same request